pkg/operators/clustermanager/controllers/certrotationcontroller/certrotation_controller.go

```diff
@@ -124,39 +95,96 @@ func (c certRotationController) sync(ctx context.Context, syncCtx factory.SyncCo
 		return nil
 	}
 
-	klog.Infof("Reconciling ClusterManager %q", clustermanagers[0].Name)
-	// do nothing if the cluster manager is deleting
-	if !clustermanagers[0].DeletionTimestamp.IsZero() {
-		return nil
-	}
+	for i := range clustermanagers {
+		clustermanagerName := clustermanagers[i].Name
 
-	// check if namespace exists or not
-	_, err = c.kubeClient.CoreV1().Namespaces().Get(ctx, helpers.ClusterManagerNamespace, metav1.GetOptions{})
-	if errors.IsNotFound(err) {
-		return fmt.Errorf("namespace %q does not exist yet", helpers.ClusterManagerNamespace)
-	}
-	if err != nil {
-		return err
-	}
+		klog.Infof("Reconciling ClusterManager %q", clustermanagerName)
+		// do nothing if the cluster manager is deleting
+		if !clustermanagers[i].DeletionTimestamp.IsZero() {
+			return nil
```

Comment:
should we update the map when the cluster manager is deleting?

Reply:
Yes, it should have a 'cleanUp' function to remove relevant secret and configmap when cluster-manager is deleted. Working on this.

Reply:
Solved, a relevant e2e test case is ongoing.